sdk/servicebus/azure-messaging-servicebus/src/main/java/com/azure/messaging/servicebus/ServiceBusSenderAsyncClient.java

```diff
@@ -371,13 +371,27 @@ public String getEntityPath() {
             return fluxError(logger, new NullPointerException("'scheduledEnqueueTime' cannot be null."));
         }
 
-        return createMessageBatch().flatMapMany(messageBatch -> {
-            messages.forEach(message -> messageBatch.tryAddMessage(message));
-            return getSendLink().flatMapMany(link -> connectionProcessor
-                .flatMap(connection -> connection.getManagementNode(entityName, entityType))
-                .flatMapMany(managementNode -> managementNode.schedule(messageBatch.getMessages(), scheduledEnqueueTime,
-                    messageBatch.getMaxSizeInBytes(), link.getLinkName(), transactionContext)));
-        });
+        return getSendLink().flatMapMany(link -> createMessageBatch()
+            .flatMapMany(messageBatch -> {
+                int index = 0;
+                for (ServiceBusMessage message : messages) {
+                    ++index;
+                    boolean added = messageBatch.tryAddMessage(message);
+                    if (!added) {
```

Comment:
nitpick only:
1. if(messageBatch.tryAddMessage(message)) is pretty readable. No need to have an additional variable if no other use.
2. Suggest index=0 instead of 1 for the first message because Java uses 0-based index.